### DISCUSSION TITLE
Apitally - Metrics, logs, and alerts for FastAPI

### DISCUSSION THREAD

**Discussion by itssimon** [fastapi/fastapi#11762] [Show and tell]
Hey FastAPI community,

When shipping APIs, having access to logs and metrics is crucial to understand API usage and troubleshoot issues. Unfortunately most monitoring tools bury these in bloated dashboards that are a pain to use.

That's why I built [Apitally](https://apitally.io/fastapi), a simple API monitoring and analytics tool for FastAPI that focuses on what actually matters for APIs:

- Usage metrics (per endpoint and consumer)
- Error tracking (with stack traces)
- Performance monitoring
- Request logs + correlated application logs
- Uptime monitoring + alerts with Slack/Teams notifications

The [Apitally SDK](https://github.com/apitally/apitally-py) integrates with FastAPI apps via a lightweight middleware, which automatically captures metrics and logs for all requests and asynchronously sends them to Apitally in regular intervals, without affecting performance.

Setup is super easy. Just add a few lines of code to your project:

```python
from fastapi import FastAPI
from apitally.fastapi import ApitallyMiddleware

app = FastAPI()
app.add_middleware(
    ApitallyMiddleware,
    client_id="your-client-id",
    env="dev",  # or "prod" etc.
)
```

There's a detailed [setup guide for FastAPI](https://docs.apitally.io/frameworks/fastapi), if you'd like to try it out.

The Apitally dashboard looks like this ⬇️

![Apitally dashboard](https://github.com/user-attachments/assets/e251aeda-85dc-46c0-9069-ade9cdcd4bde)

Please let me know what you think!

**Comment by Vincenzo140** [comment]
I like  how it’s tailored for FastAPI and keeps things simple without the overwhelm of bigger platforms. The focus on privacy is a nice touch too

**Comment by tiangolo** [comment]
Hello there, it seems you are using this post as an unfair marketing trick, by editing the same post over and over again (60 times so far) to get it to the top of the discussions.

That's not welcome here, so I'll have to close this post as spam. Please refrain from doing similar things again in the future.
